src/main/java/com/bae/service/BoulderService.java

```diff
@@ -1,5 +1,79 @@
 package com.bae.service;
 
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.bae.exceptions.BoulderNotFoundException;
+import com.bae.exceptions.InvalidDatesException;
+import com.bae.exceptions.InvalidGradeException;
+import com.bae.exceptions.InvalidStatusException;
+import com.bae.persistence.domain.Boulder;
+import com.bae.persistence.repo.BoulderRepo;
+
+@Service
 public class BoulderService {
 
+	@Autowired
+	private BoulderRepo boulderRepo;
+
+	public BoulderService(BoulderRepo boulderRepo) {
+		this.boulderRepo = boulderRepo;
+	}
+
+	public List<Boulder> getAllBoulders() {
+		return boulderRepo.findAll();
+	}
+
+	public Boulder addBoulder(Boulder newBoulder) {
+		checkCompletionDate(newBoulder);
+		checkStatus(newBoulder);
+		checkGrade(newBoulder);
+		return boulderRepo.save(newBoulder);
+	}
+
+	public String deleteBoulder(Long boulderId) {
+		if (!boulderRepo.existsById(boulderId)) {
+			throw new BoulderNotFoundException();
+		}
+		boulderRepo.deleteById(boulderId);
+		return "Boulder deleted";
+	}
+
+	public Boulder findBoulderById(Long boulderId) {
+		return boulderRepo.findById(boulderId).orElseThrow(() -> new BoulderNotFoundException());

```

Comment:
Can make this slightly neater by using:
```suggestion
		return boulderRepo.findById(boulderId).orElseThrow(BoulderNotFoundException::new);
```